### PR TITLE
test: make sure app close

### PR DESCRIPTION
**Pull request by fengmk2**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

increase mocha test timeout on coverage test

### CODE REVIEW

File: test/lib/cluster/master.test.js [hunk 1]
```diff
@@ -130,16 +130,16 @@ describe('test/lib/cluster/master.test.js', () => {
     let app2;
     before(function* () {
       mm.consoleLevel('NONE');
-      app1 = utils.cluster('apps/cluster_mod_app', { coverage: true });
-      app2 = utils.cluster('apps/cluster_mod_app', { coverage: true });
+      app1 = utils.cluster('apps/cluster_mod_app', { coverage: false });
+      app2 = utils.cluster('apps/cluster_mod_app', { coverage: false });
```
Comment (fengmk2) [inline test/lib/cluster/master.test.js]: 这些都是 egg-cluster 去覆盖的代码

**Comment by popomore** [comment]
要看看，不是所有都能关的，覆盖率下降了。

**Comment by fengmk2** [comment]
https://codecov.io/gh/eggjs/egg/compare/3d041992912d9aca1eb0edc6b226474022e08236...1368ab3d2b12579b5cf6e22174e8a273663c9d6c/changes

嗯，周一我再改改

**Comment by fengmk2** [comment]
我先补充一下覆盖率

File: package.json [hunk 2]
```diff
@@ -80,7 +80,7 @@
     "lint": "eslint app config lib test *.js",
     "test": "npm run lint -- --fix && npm run test-local",
     "test-local": "egg-bin test",
-    "cov": "egg-bin cov",
+    "cov": "TEST_TIMEOUT=100000 egg-bin cov",
```
Comment (fengmk2) [inline package.json]: 加到 100 秒，让 coverage codes 生成能跑过

File: test/lib/core/messenger.test.js [hunk 3]
```diff
@@ -55,13 +55,15 @@ describe('test/lib/core/messenger.test.js', () => {
...
     let app;
-    before(() => {
+    after(() => app.close());
+
+    // use it to record create coverage codes time
+    it('before: should start cluster app', function* () {
       app = utils.cluster('apps/messenger');
-      app.coverage(false);
-      return app.ready();
+      app.coverage(true);
```
Comment (fengmk2) [inline test/lib/core/messenger.test.js]: 这里需要开启

File: test/lib/egg.test.js [hunk 4]
```diff
@@ -151,11 +151,14 @@ describe('test/lib/egg.test.js', () => {
 
   describe('handle unhandledRejection', () => {
     let app;
-    before(() => {
+    after(() => app.close());
+
+    // use it to record create coverage codes time
+    it('before: should cluster app ready', () => {
       app = utils.cluster('apps/app-throw');
+      app.coverage(true);
```
Comment (fengmk2) [inline test/lib/egg.test.js]: 这里也是

File: package.json (hunk 2 above)
Comment (popomore) [inline package.json]: 这个 win 不行，可以 --timeout

**Comment by fengmk2** [comment]
@popomore 好了，覆盖率了涨了一点点

**Pull request merged**